plugins/org.polymap.service.geoserver/src/org/polymap/service/geoserver/GeoServerServlet.java

```diff
@@ -177,9 +198,11 @@ public void init( ServletConfig config ) throws ServletException {
         context = new BundleServletContext( getServletContext() );
         log.info( "initGeoServer(): contextPath=" + context.getContextPath() );
         
-//        sessionKey = SessionContext.current().getSessionKey();
-//        assert sessionKey != null;
-
+        //String sessionKey = SessionContext.current().getSessionKey();
+        //assert sessionKey != null;
+        sessionKey = "ows_" + System.currentTimeMillis();
+        contextProvider.mapContext( sessionKey, true );
```

Comment:
SessionContext wird an mehreren Stellen benötigt. 

UIJob bspw. brauch den, um den Job überhaupt auszuführen. Das war noch leicht zu finden, da gleich ne NPE gekommen. UIJob könnte ich auch durch Future oder irgend ne andere Threadimplementierung ersetzen, aber so gefällt es mir eigentlich ganz gut.

Kritischer ist das Laden der Features. Dort wird org.polymap.core.data.AsyncPipelineFeatureCollection.AsyncPipelineIterator.AsyncPipelineIterator(AsyncPipelineFeatureCollection) benutzt und dieser wiederum führt den fetcher nur aus wenn auch ein SessionContext vorhanden ist, Zeile 197ff. Und da wüsste ich nicht, wie ich das umkonfiguriere/ändere.

Was mir halt komisch war, ist dass das Laden im SingleLayer-Fall immer geht, also auch ohne SessionContext und im MultiLayer-Fall geht es einmal ohne SessionContext und beim 2. Request bleibt es im  AsyncPipelineIterator.hasNext() hängen, weil kein Wert mehr aus der BlockingQueue zur Verfügung steht.

So wie es jetzt implementiert ist, funktioniert es, aber es ist wie Du sagst, ich wundere mich, warum es geht, weil ich den Context ja nur für den Thread gemacht habe, in dem das *Geoservlet.init()* aufgerufen wird. Korrekt wäre es, das für alle *service* calls zu machen, also in *inContextClassLoader*